lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,946 @@
+            set_up_cube())
+        coord = DimCoord(0.5, long_name="threshold", units='mm hr^-1')
+        self.cube.add_aux_coord(coord)
+        self.cube.add_cell_method(CellMethod('mean', coords='realization'))
+
+    def test_basic(self):
+        """Test that the method returns the expected cube type
+        and that the metadata are as expected.
+        We expect a new name, the threshold coord to be removed and
+        cell methods to be discarded."""
+        plugin = Plugin()
+        result = plugin._update_metadata(self.cube)
+        self.assertIsInstance(result, Cube)
+        self.assertEqual(result.name(), "probability_of_lightning")
+        msg = ("Expected to find exactly 1 threshold coordinate, but found "
+               "none.")
+        with self.assertRaisesRegex(CoordinateNotFoundError, msg):
+            result.coord('threshold')
+        self.assertEqual(result.cell_methods, ())
+
+    def test_input(self):
+        """Test that the method does not modify the input cube data."""
+        plugin = Plugin()
+        incube = self.cube.copy()
+        plugin._update_metadata(incube)
+        self.assertArrayAlmostEqual(incube.data, self.cube.data)
+        self.assertEqual(incube.metadata, self.cube.metadata)
+
+    def test_missing_threshold_coord(self):
+        """Test that the method raises an error in Iris if the cube doesn't
+        have a threshold coordinate to remove."""
+        self.cube.remove_coord('threshold')
+        plugin = Plugin()
+        msg = ("Expected to find exactly 1 threshold coordinate, but found no")
+        with self.assertRaisesRegex(CoordinateNotFoundError, msg):
+            plugin._update_metadata(self.cube)
+
+
+class Test__modify_first_guess(IrisTest):
+
+    """Test the _modify_first_guess method."""
+
+    def setUp(self):
+        """Create cubes with a single zero prob(precip) point.
+        The cubes look like this:
+     precipitation_amount / (kg m^-2)
+     Dimension coordinates:
+        time: 1;
+        projection_y_coordinate: 16;
+        projection_x_coordinate: 16;
+     Auxiliary coordinates:
+          forecast_period (on time coord): 4.0 hours (simulates UM data)
+     Scalar coordinates:
+          forecast_reference_time: 2015-11-23 03:00:00
+     Data:
+       self.cube:
+          forecast_period (on time coord): 0.0 hours (simulates nowcast data)
+          All points contain float(1.) except the
+          zero point [0, 0, 7, 7] which is float(0.)
+       self.fg_cube:
+          All points contain float(1.)
+       self.ltng_cube:
+          forecast_period (on time coord): 0.0 hours (simulates nowcast data)
+          All points contain float(1.)
+       self.precip_cube:
+          With extra coordinate of length(3) "threshold" containing
+          points [0.5, 7., 35.] mm hr-1.
+          All points contain float(1.) except the
+          zero point [0, 0, 7, 7] which is float(0.)
+          and [1:, 0, ...] which are float(0.)
```

Comment:
And we're 100% certain of getting > 35 mm precipitation everywhere, except at one spatial point (which coordinate is threshold)?

Reply:
That's right, so that for all points except the one we're varying, nothing changes from the first-guess field to the output field.